src/app/containers/ConsentBanner/index.canonical.jsx

```diff
@@ -18,9 +19,14 @@ const Canonical = () => {
 
   useEffect(runInitial, []);
 
-  const onCookieAccept = () => {
-    cookieOnAllow();
-    updateCookiePolicy();
+  const onCookieAction = actionType => {
+    if (actionType === 'accept') {
+      cookieOnAllow();
+      updateCookiePolicy();
+    } else if (actionType === 'reject') {
+      cookieOnReject();
+    }
+    onDismissFocusRef?.current?.querySelector('a')?.focus();
```

Comment:
Seperate comment on this, I see there is a `querySelector('a')` here, can the `ref` refer directly to the anchor tag rather than the wrapping element? This means the ref you pass into the consent banner must always contain a `a` tag in order for the focus behaviour to work.

Reply:
Thanks for the comments @andrewscfc!
Ah! Nice spot, must've got confused with amp. I'll ensure it only pulls focus on accept, but I'll keep the `onCookieAction` method for future use 👍

Regarding the `querySelector('a')`, a problem with using the ref directly is that the underlying structure of `Brand` contains a lot of conditional rendering, and so the ref itself wouldn't always be defined if it was placed on a deeper nested dom element

Reply:
We put a lot of thought into this - and the thinking behind this is that the anchor is a conditional element, sometimes it could be there, sometimes it might not. We thought it was best to provide a stable ref than to provide one who could become undefined on another render. Instead we are being explicit about what we want to select at a given moment, whilst knowing we have a stable parent element to construct our DOM calls from.

Reply:
Okay, so what happens if the `a` element is undefined, it doesn't focus on anything? If so, code like this would be functionally equivalent if the anchor tag had the ref applied?
```
    onDismissFocusRef?.current?.focus();
```

Reply:
Brand is a reusable component. If you engineer it as suggested, ref will be useless in the non-anchored state. As it stands, it will always provide a stable ref. It is then up to the engineer what they want to do with it after they've plugged it into their application.

Reply:
Sorry I'm not quite following tbh, I'm happy to have a call about this or just move on as it isn't a huge issue.